v3/src/components/case-table/case-table-title-bar.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from "react"
+import { ComponentTitleBar, EditableComponentTitle  } from "../component-title-bar"
+import { Box, CloseButton, Flex } from "@chakra-ui/react"
+import t from "../../utilities/translation/translate"
+import MinimizeIcon from "../../assets/icons/icon-minimize.svg"
+import TableIcon from "../../assets/icons/icon-table.svg"
+import CardIcon from "../../assets/icons/icon-case-card.svg"
+import { useDataSetContext } from "../../hooks/use-data-set-context"
+
+export const CaseTableTitleBar = () => {
+  const dataset = useDataSetContext()
+  const [title, setTitle] = useState(dataset?.name || "Dataset")
```

Comment:
If the user changes the title that probably needs to be saved somewhere, so this value should be set in the model not just in React state. But that can be a feature for another day if you don't want to deal with it now.

Reply:
I'll write a story for it so we remember to implement it later. Story is [here](https://www.pivotaltracker.com/story/show/184374934)